elastic-stack/elasticsearch/configure.sls

```diff
@@ -84,11 +84,17 @@ increase_elasticsearch_file_descriptor_limit:
     - name: sysctl -w fs.file-max={{ elasticsearch.fd_limit }}
   file.replace:
     - name: /etc/sysctl.conf
-    - pattern: '^fs.file_max=.*'
-    - repl: fs.file_max={{ elasticsearch.fd_limit }}
+    - pattern: '^fs.file-max=.*'
+    - repl: fs.file-max={{ elasticsearch.fd_limit }}
     - append_if_not_found: True
     - onchanges_in:
         - service: elasticsearch_service
+  file.replace:
+    - name: /usr/lib/systemd/system/elasticsearch.service
```

Comment:
Systemd supports creating template fragments to override existing units by placing them in `/etc/systemd/system/<unit name>` so we can just create the fragment there to set the file limit. Also, when editing systemd units it's generally necessary to run `systemctl daemon-reload`

Reply:
I've added commit aca4d8d, which changes this state to write out that `/etc/systemd/system` unit file. This appears to override anything that's set by `systemctl` so I think that this is the only action necessary.

Reply:
Oops. I think I need to change this. I missed the link you pasted below to the serverfault.com page.